### PR TITLE
Niad 2935 remove fallback

### PR DESCRIPTION
**Pull request by Alex-Nita**
## What

Removed fallback option for filling out authoredOn field in JSON.

## Why

Previously, it was relying on AvailabilityTime, that cause issues with the field being updated unnecessarily when AvailabilityTime was updated.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Internal change (non-breaking change with no effect on the functionality affecting end users)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have updated the [Changelog](/CHANGELOG.md) with details of my change in the UNRELEASED section if this change will affect end users
- [ ] A corresponding change has been made to the [Mapping Documentation repository][mapping-docs]

[mapping-docs]: https://github.com/NHSDigital/patient-switching-adaptors-mapping-documentation

### CODE REVIEW

File: gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ProcedureRequestMapper.java [hunk 1]
```diff
@@ -60,7 +59,7 @@ public ProcedureRequest mapToProcedureRequest(
         procedureRequest
             .setStatus(ProcedureRequestStatus.ACTIVE)
             .setIntent(ProcedureRequestIntent.PLAN)
-            .setAuthoredOnElement(getAuthoredOn(planStatement.getAvailabilityTime(), ehrComposition))
```
Comment (adrianclay) [inline gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ProcedureRequestMapper.java:63]: Looking [at the spreadsheet](https://docs.google.com/spreadsheets/d/1pG59D2Et3m7i7ne3a8Le7b_JhEOPwehZCG2FUMLRdDA/edit#gid=0) I'm confused as to why we changed `ProcedureRequestMapper`?

File: gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ReferralRequestMapper.java [hunk 2]
```diff
@@ -77,7 +76,7 @@ public ReferralRequest mapToReferralRequest(RCMRMT030101UKEhrComposition ehrComp
         var id = requestStatement.getId().get(0).getRoot();
         var identifier = buildIdentifier(id, practiseCode);
         var agent = ParticipantReferenceUtil.getParticipantReference(requestStatement.getParticipant(), ehrComposition);
-        var authoredOn = getAuthoredOn(requestStatement.getAvailabilityTime());
+        var authoredOn = getAuthoredOn(ehrComposition);
```
Comment (adrianclay) [inline gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ReferralRequestMapper.java:79]: I'm also confused as to why the `ReferralRequestMapper` has changed?

File: gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapper.java [hunk 3]
```diff
@@ -61,7 +60,7 @@ private Stream<DomainResource> mapMedicationStatement(RCMRMT030101UKEhrExtract e
 
         var context = encounters.stream()
             .filter(encounter1 -> encounter1.getId().equals(ehrComposition.getId().getRoot())).findFirst();
-        var authoredOn = getAuthoredOn(medicationStatement.getAvailabilityTime(), ehrExtract, ehrComposition);
```
Comment (adrianclay) [inline gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapper.java:64]: So the idea behind this ticket was to remove the fallback, but it appears you've removed the primary time field. Am I confused?

File: gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapperTest.java [hunk 4]
```diff
@@ -230,7 +227,82 @@ public void When_MappingMedicationRequestWithAvailabilityTimeOnlyInEhrExtract_Ex
...
+                .filter(resource -> ResourceType.MedicationRequest.equals(resource.getResourceType()))
+                .map(MedicationRequest.class::cast)
+                .findFirst()
+                .get();
+
+        assertThat(medicationRequest.getAuthoredOnElement().getValue()).isEqualTo(expectedAuthoredOn.getValue());
+    }
+
+    @Test
+    public void When_MappingMedicationRequestWithAuthoredOnValidDateInExtractOnly_Expect_AuthoredOnToUseExtractValue() {
```
Comment (adrianclay) [inline gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapperTest.java:284]: I'm a bit confused by this test as the title of the test says "InExtractOnly", but the test appears to be using a date in the `MedicationStatement` element.

File: gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapperTest.java [hunk 5]
```diff
@@ -230,7 +227,82 @@ public void When_MappingMedicationRequestWithAvailabilityTimeOnlyInEhrExtract_Ex
...
+                .filter(resource -> ResourceType.MedicationRequest.equals(resource.getResourceType()))
+                .map(MedicationRequest.class::cast)
+                .findFirst()
+                .get();
+
+        assertThat(medicationRequest.getAuthoredOnElement().getValue()).isEqualTo(expectedAuthoredOn.getValue());
+    }
+
+    @Test
+    public void When_MappingMedicationRequestWithAuthoredOnValidDateInExtractAndComposition_Expect_AuthoredOnToUseCompositionValue() {
```
Comment (adrianclay) [inline gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapperTest.java:259]: Is this test also just using the value from the `MedicationStatement` too or am I being thick?

**Review by adrianclay — COMMENTED** [review]
I'm a tad surprised there aren't any test changes for `AllergyIntoleranceMapper`.

File: gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapperTest.java (hunk 5 above)
Comment (Alex-Nita) [reply gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapperTest.java]: Title was outdated as well.

**Review by adrianclay — APPROVED** [review]

**Pull request merged**